lib/Optimizer/Transforms/GenKernelExecution.cpp

```diff
@@ -796,35 +796,35 @@ class GenerateKernelExecution
           builder.setInsertionPointToStart(module.getBody());
 
           // Create the function if it doesn't already exist.
-          if (!module.lookupSymbol<LLVM::LLVMFuncOp>(cudaqRegisterLambdaName))
+          if (!module.lookupSymbol<LLVM::LLVMFuncOp>(cudaqRegisterCallableName))
             builder.create<LLVM::LLVMFuncOp>(
-                module.getLoc(), cudaqRegisterLambdaName,
+                module.getLoc(), cudaqRegisterCallableName,
                 LLVM::LLVMFunctionType::get(
                     cudaq::opt::factory::getVoidType(ctx),
                     {cudaq::opt::factory::getPointerType(ctx),
                      cudaq::opt::factory::getPointerType(ctx)}));
 
-          // Create this global name, it is unique for any lambda
+          // Create this global name, it is unique for any callable
           // bc classNameStr contains the parentFunc + varName
-          auto lambdaName = builder.create<LLVM::GlobalOp>(
+          auto callableName = builder.create<LLVM::GlobalOp>(
               loc,
               cudaq::opt::factory::getStringType(ctx, demangledName.size() + 1),
               /*isConstant=*/true, LLVM::Linkage::External,
-              classNameStr + ".lambdaName",
+              classNameStr + ".callableName",
               builder.getStringAttr(demangledName + '\0'), /*alignment=*/0);
 
           builder.restoreInsertionPoint(insertPoint);
-          auto lambdaRef = builder.create<LLVM::AddressOfOp>(
-              loc, cudaq::opt::factory::getPointerType(lambdaName.getType()),
```

Comment:
Yeah, this file we can leave "lambda".

FWIW, other callables have names given to them by the programmer. We have to go to extra lengths for lambda expressions.